File: app/src/main/java/com/tichuguru/db/HandDao.java
package com.tichuguru.db;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface HandDao {
    @Query("SELECT * FROM hands WHERE gameId = :gameId ORDER BY handOrder")
    List<HandEntity> getHandsForGame(long gameId);

    @Insert
    void insertAll(List<HandEntity> hands);

    @Query("DELETE FROM hands WHERE gameId = :gameId")
    void deleteHandsForGame(long gameId);

    @Query("DELETE FROM hands")
    void deleteAll();
}

File: app/src/main/kotlin/com/tichuguru/db/HandDao.kt
package com.tichuguru.db

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface HandDao {
    @Query("SELECT * FROM hands WHERE gameId = :gameId ORDER BY handOrder")
    fun getHandsForGame(gameId: Long): List<HandEntity>

    @Upsert
    fun upsertAll(hands: List<HandEntity>): List<Long>

    @Query("DELETE FROM hands WHERE gameId = :gameId")
    fun deleteHandsForGame(gameId: Long)

    fun deleteOrphanHands(
        gameId: Long,
        keepIds: List<Long>,
    ) {
        if (keepIds.isNotEmpty()) deleteOrphanHandsInternal(gameId, keepIds)
    }

    @Query("DELETE FROM hands WHERE gameId = :gameId AND id NOT IN (:keepIds)")
    fun deleteOrphanHandsInternal(
        gameId: Long,
        keepIds: List<Long>,
    )
}
